template_handler/cdk.py

```diff
@@ -0,0 +1,218 @@
+        return cmd_exists
+
+    def _node_package_exists(self, package) -> bool:
+        """
+        Checks whether a specifie Node package exists either in the workspace or global scope
+        """
+
+        package_exists = False
+        workspace_result = subprocess.run(
+            f'npm list {package}',
+            shell=True,
+            capture_output=True
+        )
+        self.logger.debug(f"Workspace NPM package '{package}' exists: {not bool(workspace_result.returncode)}")
+
+        if workspace_result.returncode == 0:
+            package_exists = True
+        else:
+            global_result = subprocess.run(
+                f'npm --global list {package}',
+                shell=True,
+                capture_output=True
+            )
+            self.logger.debug(f"f'Global NPM package '{package}' exists: {not bool(global_result.returncode)}")
+
+            if global_result.returncode == 0:
+                package_exists = True
+
+        return package_exists
+
+    def _check_prerequisites(self) -> None:
+        """
+        Checks the command and Node package requirements for the handler
+        """
+
+        # Check Command Prerequisites
+        cmd_prerequisites = [
+            'node',
+            'npx'
+        ]
+        for cmd_prerequisite in cmd_prerequisites:
+            if not self._cmd_exists(cmd_prerequisite):
+                raise exceptions.SceptreException(f"{self.name} - Command prerequisite '{cmd_prerequisite}' not found")
+
+        # Check Node Package Prerequisites
+        node_prerequisites = [
+            'cdk-assets'
+        ]
+        for node_prerequisite in node_prerequisites:
+            if not self._node_package_exists(node_prerequisite):
+                raise exceptions.SceptreException(f"{self.name} - Node Package prerequisite '{node_prerequisite}' not found")
+
+    def handle(self) -> str:
+        """
+        Main Sceptre CDK Handler function
+
+        Returns
+        -------
+        str|bytes
+            CDK synthesised CloudFormation template
+        """
+
+        self.logger = self.PrefixLoggerAdapter(self.logger, {'prefix': self.name})
+
+        self._check_prerequisites()
+
+        # Import CDK Python template module
+        template_path = str(pathlib.Path('templates', self.cdk_template_path))
+        self.logger.debug(f'Importing CDK Python template module {template_path}')
+        template_module_name = pathlib.Path(template_path).stem
+        loader = importlib.machinery.SourceFileLoader(template_module_name, template_path)
+        spec = importlib.util.spec_from_loader(template_module_name, loader)
+        template_module = importlib.util.module_from_spec(spec)
+        loader.exec_module(template_module)
+
+        # CDK Synthesize App
+        self.logger.debug(f'CDK synthesing CdkStack Class')
+        self.logger.debug(f'CDK Context: {self.cdk_context}')
+        app = aws_cdk.App(context=self.cdk_context)
```

Comment:
Just double-checking here: Since `self.cdk_context` might very well be `None`, is it safe to pass `None` for context? Does that automatically default to the default context? I'm only asking because I see the possibility that `None` could instead communicate "don't use any context". I'm not super familiar with CDK's python api.